content/docs/guides/crosswalk/kubernetes/aws/02-kubernetes-stacks/01-cluster-configuration.md

```diff
@@ -0,0 +1,672 @@
+---
+
+title: Kubernetes Cluster
+
+aliases: ["/docs/guides/crosswalk/kubernetes/aws/cluster/"]
+---
+
+Manage the control plane, worker nodes, RBAC, and defaults.
+
+  * [Create the Control Plane](#create-the-control-plane)
+  * [Create the Worker Nodes](#create-the-worker-nodes)
+  * [Use the Cluster](#use-the-cluster)
+  * [Setting Defaults](#setting-defaults)
+  * [Use Kuberntes RBAC](#use-kubernetes-rbac)
+  * [Updating the Worker Nodes](#updating-the-worker-nodes)
+
+
+The full code for this stack is on [GitHub][gh-repo-stack].
+
+See the [official Kubernetes docs][k8s-docs] for more details.
+
+## Create the Control Plane
+
+Create a new EKS cluster control plane equipped with:
+
+**Identity**
+
+  * IAM roles for each worker node group.
+  * IAM roles for users mapped into Kubernetes RBAC.
+
+**Networking**
+
+  * Public subnets for use with public load balancers.
+  * Private subnets for use with private load balancers.
+  * Private subnets used as the default subnets for workers, by design.
+  * Pod networking managed by the [AWS CNI Plugin][aws-k8s-cni].
+
+**Storage**
+
+  * Default Kubernetes storage classes backed by cloud volume types.
+
+**Workers**
+
+  * No default worker node group in favor of specific, separately managed node groups.
+
+**General**
+
+  * A specific version of Kubernetes for the control plane.
+  * Tags for resources under cluster management.
+  * Control plane logging enabled.
+  * Configurable accessibility for the cluster API endpoint (optional).
+
+    ```typescript
+    import * as eks from "@pulumi/eks";
+    
+    // Create an EKS cluster.
+    const cluster = new eks.Cluster(`${projectName}`, {
```

Comment:
Is a properly configured default cluster something we are considering adding to Kx?

Reply:
We had not considered this given that not all clusters will be configured identically e.g. IAM, node groups, tags etc. There are benefits to a better default cluster OOTB such as removing boilerplate, copy/paste etc, but it feels like this will start to encroach on an opinionated installer. Thoughts @lblackstone?

Reply:
I definitely understand, and it is very possible this isn't the right thing to pursue. But "semi-opinionation" is exactly the name of the game for the X-family of packages. (E.g., that's what makes the AWSX VPC so special -- it's the right set of defaults for 80% of the users, and helps to tame an otherwise very difficult space that would require consulting documentation otherwise.) The trick is ensuring the right knobs are present, but I do wonder if there is something magical we can do here (and, for GCP, we could get rid of that gnarly kubeconfig string interpolation that we unfortunately need to put into example...)